nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/data/explorer/PutAzureDataExplorer.java

```diff
@@ -0,0 +1,309 @@
+            .dependsOn(POLL_ON_INGESTION_STATUS, "true")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("600")
+            .build();
+
+    static final PropertyDescriptor INGESTION_STATUS_POLLING_INTERVAL = new PropertyDescriptor.Builder()
+            .name("Ingestion status polling interval")
+            .displayName("Ingestion status polling interval in seconds")
+            .description("Defines the value of timeout for polling on ingestion status in seconds.")
+            .required(false)
+            .dependsOn(POLL_ON_INGESTION_STATUS, "true")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("5")
+            .build();
+
+    public static final Relationship SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Relationship for success")
+            .build();
+    public static final Relationship FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Relationship for failure")
+            .build();
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptorList = new ArrayList<>();
+        descriptorList.add(ADX_SERVICE);
+        descriptorList.add(DATABASE_NAME);
+        descriptorList.add(TABLE_NAME);
+        descriptorList.add(MAPPING_NAME);
+        descriptorList.add(DATA_FORMAT);
+        descriptorList.add(IGNORE_FIRST_RECORD);
+        descriptorList.add(IS_STREAMING_ENABLED);
+        descriptorList.add(POLL_ON_INGESTION_STATUS);
+        descriptorList.add(ROUTE_PARTIALLY_SUCCESSFUL_INGESTION);
+        descriptorList.add(INGESTION_STATUS_POLLING_TIMEOUT);
+        descriptorList.add(INGESTION_STATUS_POLLING_INTERVAL);
+        this.descriptors = Collections.unmodifiableList(descriptorList);
+
+        final Set<Relationship> relationshipSet = new HashSet<>();
+        relationshipSet.add(SUCCESS);
+        relationshipSet.add(FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationshipSet);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        service = context.getProperty(ADX_SERVICE).asControllerService(KustoIngestService.class);
+        if (checkIfIngestorRoleDoesntExist(context.getProperty(DATABASE_NAME).getValue(), context.getProperty(TABLE_NAME).getValue())) {
+            getLogger().error("User might not have ingestor privileges, table validation will be skipped for all table mappings.");
+            throw new ProcessException("User might not have ingestor privileges, table validation will be skipped for all table mappings. ");
```

Comment:
It is not necessary to both log an error and throw an exception. Recommend removing the error log and include the database and table name in the exception message.

The message says that validation will be skipped, but throwing an exception prevents the Processor from starting, so the behavior should be clarified.

Reply:
Since without ingestor privilege, there is no point in starting the ingestion of data, hence we throw an exception. Also removed skipping of table validation message which might confuse the user.